systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java

```diff
@@ -528,15 +527,15 @@ protected String getResourceApiVersion(String resourcePlural, String coVersion)
         }
     }
 
-    protected JsonObject getConversionToolDataFromUpgradeJSON() {
-        JsonArray upgradeFile = readUpgradeJson(UPGRADE_JSON_FILE);
-        return upgradeFile.getJsonObject(0).getJsonObject("conversionTool");
+    protected Map<String, String> getConversionToolDataFromUpgradeYAML() {
+        List<VersionModificationData> versionModificationData = getVersionModificationData(UPGRADE_YAML_FILE);
+        return versionModificationData.get(0).getConversionTool();
     }
 
-    protected void convertCRDs(JsonObject conversionTool, String namespace) throws IOException {
-        String url = conversionTool.getString("urlToConversionTool");
+    protected void convertCRDs(Map<String, String> conversionTool, String namespace) throws IOException {
+        String url = conversionTool.get("urlToConversionTool");
```

Comment:
Can you have extra structure/class just for the conversion tool? It would simplify the getters